documentation/resource-definitions.md

```diff
@@ -0,0 +1,118 @@
+# Resource definitions 
+
+## V1 version
+If V1 version of Resource is needed, please add the following in the `definitions` section of swagger spec.
+
+```
+	"Resource": {
+          "properties": {
+            "id": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource Id"
+            },
+            "name": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource name"
+            },
+            "type": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource type"
+            },
+            "location": {
+              "type": "string",
+              "description": "Resource location"
+            },
+            "tags": {
+              "type": "object",
+              "additionalProperties": {
+                "type": "string"
+              },
+              "description": "Resource tags"
+            }
+          },
+          "required": [
```

Comment:
Location is not required for child resources, in case any RP is having child resources inherited from resource, they may need to remove this and add it to specific resource definition or need to override this property (if it can be). I think we should put this information in this file as well.